system-addon/lib/PlacesFeed.jsm

```diff
@@ -100,13 +100,19 @@ class BookmarksObserver extends Observer {
    * @param  {str} title
    * @param  {int} dateAdded
    * @param  {str} guid      The unique id of the bookmark
+   * @param  {str} parent guid
+   * @param  {int} source    Used to distinguish bookmarks made by different
+   *                         actions: sync, bookmarks import, other.
    */
   onItemAdded(...args) {
     const type = args[3];
-    if (type !== PlacesUtils.bookmarks.TYPE_BOOKMARK) {
+    const source = args[9];
+    const uri = args[4];
+    if (type !== PlacesUtils.bookmarks.TYPE_BOOKMARK ||
+        source !== PlacesUtils.bookmarks.SOURCES.DEFAULT ||
+        (uri && (uri.scheme !== "http" && uri.scheme !== "https"))) {
```

Comment:
This check for `uri` shouldn't be necessary as if we're getting a `TYPE_BOOKMARK`, it should definitely have a `uri`.

Reply:
@Mardak the very first event that comes in to create the `Firefox Nightly Resources` folder does not have an uri.

Reply:
The `||` operator short circuits, so something like `true || null.foo` does not throw. The `uri.scheme` won't throw an exception for "Firefox Nightly Resources", which is both *not* a `TYPE_BOOKMARK` and *is* a `SOURCES.IMPORT_REPLACE`. That's why I said a `TYPE_BOOKMARK` should definitely have a `uri`.

Reply:
Here's more info https://en.wikipedia.org/wiki/Short-circuit_evaluation

Reply:
Or javascript specific on MDN:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Short-circuit_evaluation

Reply:
Or if you were confused by the precedence of things: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence